---
title: "Visualizing_Alignments"
author: "Caden McQuillen"
date: '2023-03-14'
output: html_document
---

## Genome and Fastq Info
1. hg38 genome
2. gencode v41 annotation
3. Single-ended


## Get ACTB genomic coordiants
In hg38 genome, ACTB coordinates  are: chr7:5,525,148-5,532,601

## Filter BAMs for ACTB gene
```{bash, eval = FALSE}
mamba activate angsd
samtools view -b SRR11849597.Aligned.sortedByCoord.out.bam  chr7:5525148-5532601 > SRR11849597.actb.bam
samtools index SRR11849597.actb.bam

```

## Index and zip GTF file
```{bash, eval = FALSE}
mamba activate angsd
cat hg38.gencodev41.gtf | sort -S4G -k1,1 -k4,4n | bgzip -c > hg38.gencodev41.sorted.gtf.gz
tabix hg38.gencodev41.sorted.gtf.gz

```

## IGV Png
![IGV ACTB ](/Users/Caden/Desktop/angsd/project/ACTB_hg38.png)

## Three Observations
1. My data appears to be unstranded since we have reads going in both directions for the same gene. 
2. My data appears to be ribo depleted since we have reads aligning to intronic regions. 
3. Interestingly there appears to be a refseq feature that a decent amount of reads are aligning to that is not present in the gencode annotations. From a quick google search it appears to be an un-characterized lncRNA. 
